### PR TITLE
add support for las points 6 and 8 and bug fix for points 3 and 7

### PR DESCRIPTION
**Pull request by amir-enzen**

### CODE REVIEW

File: Manifest.toml [hunk 1]
```diff
@@ -1,330 +1,347 @@
 # This file is machine-generated - editing it directly is not advised
```
Comment (greghislop) [inline Manifest.toml]: is there an advantage in changing the Manifest,? maybe I should never have included it? perhaps not include in your PR?

File: src/pointcloud_io.jl [hunk 2]
```diff
@@ -354,7 +354,7 @@ function make_table(points::AbstractVector{LasPoint3}, offset, scale)
     classification = map(p -> p.raw_classification, points)
     pointsourceid = map(p -> p.pt_src_id, points)
     gpstime = map(p -> p.gps_time, points)
-    userdata = map(p -> p.user_data, points)
+    # userdata = map(p -> p.user_data, points)
```
Comment (greghislop) [inline src/pointcloud_io.jl:357]: remove

Comment (greghislop) [reply src/pointcloud_io.jl:357]: same below old code

Comment (greghislop) [reply src/pointcloud_io.jl:357]: actually why are you remioving the userdata?

File: src/pointcloud_io.jl [hunk 3]
```diff
@@ -403,6 +428,37 @@ function make_table(points::AbstractVector{LasPoint7}, offset, scale)
...
+    end
+
+    intensity = map(p -> p.intensity, points)
+    returnnumber = map(p -> p.flag_byte_1 & 0b00001111, points)
+    numberofreturns = map(p -> (p.flag_byte_1 & 0b11110000) >> 4, points)
+    classification = map(p -> p.classification, points)
+    pointsourceid = map(p -> p.pt_src_id, points)
+    gpstime = map(p -> p.gps_time, points)
+    userdata = map(p -> p.user_data, points)
+    nir = map(p -> p.nir, points)
```
Comment (greghislop) [inline src/pointcloud_io.jl:443]: did you test this on a point format 8 file? if not that's fine just asking

File: src/pointcloud_io.jl [hunk 4]
```diff
@@ -530,76 +586,106 @@ function save_las(filename::AbstractString, pc::AbstractVector{<:NamedTuple}; x_
...
                             point_return_count[p.returnnumber] += 1
                         else
                             point_return_count[1] += 1
                         end
                     end
                 end
-            else
-                data_format_id = 1
-                data_record_length = 28
-                data = Vector{LasPoint1}(undef, records_count)
```
Comment (greghislop) [inline src/pointcloud_io.jl:565]: I don't know why you are changing the earlier formats of 0->3 and haven't checked if it's right, not sure it matters thoes would be old files now

**Review by greghislop — APPROVED** [review]
have a look at my comments and deal with them or not as you see fit, thanks

File: Manifest.toml (hunk 1 above)
Comment (amir-enzen) [reply Manifest.toml]: It was suggested by julia 1.7 but i'll revert it anyways!

File: src/pointcloud_io.jl (hunk 2 above)
Comment (amir-enzen) [reply src/pointcloud_io.jl:357]: That's about the problem I mentioned, If we leave user data we can't write las point 3 anymore, all will be 7.

File: src/pointcloud_io.jl (hunk 3 above)
Comment (amir-enzen) [reply src/pointcloud_io.jl:443]: yes!

File: src/pointcloud_io.jl (hunk 4 above)
Comment (amir-enzen) [reply src/pointcloud_io.jl:565]: I added some code and git thinks I have changed it but it only has been pushed around.

**Pull request merged**